source/1_engine/rex_engine/include/rex_engine/filesystem/vfs.h

```diff
@@ -0,0 +1,70 @@
+#pragma once
+
+#include "rex_engine/filesystem/mount_roots.h"
+#include "rex_engine/types.h"
+#include "rex_std/bonus/functional.h"
+#include "rex_std/functional.h"
+#include "rex_std/memory.h"
+#include "rex_std/string_view.h"
+#include "rex_std/type_traits.h"
+#include "rex_std_extra/utility/type_id.h"
+#include "rex_std_extra/utility/yes_no.h"
+
+namespace rex::vfs
```

Comment:
I don't like this style of namespaces, could we add nested ones?
I think we already do nested ones in different places.
This adds to readability as well cuz I can see the indentation an I know that this code is included within more namespaces than just "rex"